graph/db/graph_sql_test.go

```diff
@@ -0,0 +1,118 @@
+//go:build test_db_postgres || test_db_sqlite
+
+package graphdb
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestNodeIsPublicCache verifies that once a node is observed as public, it
+// remains public on cache hit even if it later has zero public channels in the
+// DB.
```

Comment:
can you add a comment here explaining that once a public key is leaked we do not invalidate the key from the cache even if it has not public channels anymore because once leaked the node cannot undo this state.